PSKoans/Public/Show-Karma.ps1

```diff
@@ -80,6 +85,29 @@ function Show-Karma {
                 $KoanLocation | Invoke-Item
             }
         }
+        'OpenFile' {
+            try {
+                $Results = Get-Karma @GetParams
+            }
+            catch {
+                $PSCmdlet.ThrowTerminatingError($_)
+            }
+            $FilePath = Get-PSKoan -Topic $Results.CurrentTopic.Name -Scope User | Select-Object -ExpandProperty Path
+            $LineNumber = $Results.CurrentTopic.FailedLineNumber
+            if ($PSVersionTable.Platform -eq "win32nt") {
```

Comment:
This works great in Windows, but we need to do the same for Linux, for now, I just open the file with the default editor, any suggestion is appreciated. 😊

Reply:
Hmm. Well, we can't use this on Windows PowerShell; `Platform` was added sometime in PS6+

Also, this won't respect the `Editor` setting from PSKoans itself (`Get-PSKoanSetting -Name Editor`); that should probably be all we check tbh. It's possible we should have a more flexible / system-aware default setting, but I think that would be a separate PR entirely and would be taken care of outside this command. All this needs to do is query the selected editor, which the user can set at any time with `Set-PSKoanSetting`.

Reply:
I don't see why this wouldn't work on Unix OSes? Starting a process generally works about the same on both OSes, minus some edge cases with some issues around quoting command line arguments... but I don't think we're going to hit those there.

Reply:
I was talking about the registry line. 😊 I don't have much Linux experience but for sure the registry key extraction of the default editor name won't work in Linux. So probably there should be a file or something to extract the default program name for a specific extension in Linux.

Reply:
Yeah, that's not something I'd expect this command to handle, I don't think. We can look into possibly examining that when we set the initial default setting, but I don't think we should be overriding a setting that may come from the user here. 🙂

Reply:
There is another problem that I couldn't figure out how to resolve. Atom, VSCode, PowerShell_ISE they all have a registry key or environment variable I think that makes them easy to run (by just typing their name in the terminal) but this is not the case with "Sublime Text 3" and some other editors. Either we need to make them run like the others by adding a key or variable or use a completely different way to be able to cover more IDEs. Any suggestion? 🤔

Reply:
That's not something we can handle, really. The editors that do work are adding the path to the folder their executables are in to the $env:PATH variable; if folx want to do that themselves they can make that work with [Environment]::SetEnvironmentVariable(), but tbh messing with the path variables is something that we probably shouldn't delve into here. 

We can always add something to the documentation to explain how to add folders to the PATH variable if it becomes an issue. 🙂